omod/src/test/webapp/resources/scripts/navigatorModels.js

```diff
@@ -219,13 +219,14 @@ describe("Test for simple form models", function() {
 
         beforeEach(function() {
             questionModel = new QuestionModel();
-            firstField = jasmine.createSpyObj('firstField', ['unselect', 'resetErrorMessages', 'value', 'displayValue', 'enable', 'disable', 'hide', 'show']);
+            firstField = jasmine.createSpyObj('firstField', ['unselect', 'resetErrorMessages', 'value', 'displayValue', 'enable', 'disable', 'hide', 'show', 'isValid']);
             secondField = jasmine.createSpyObj('secondField', ['unselect', 'resetErrorMessages', 'value', 'displayValue', 'enable', 'disable', 'hide', 'show']);
             firstField.element = jasmine.createSpyObj('element', ['addClass', 'removeClass', 'hasClass']);
             secondField.element  = jasmine.createSpyObj('element', ['addClass', 'removeClass', 'hasClass']);
             questionModel.fields = [firstField, secondField];
             element = jasmine.createSpyObj('element', ['addClass', 'removeClass', 'triggerHandler', 'attr', 'removeAttr', 'hide', 'show']);
             questionModel.element = element;
+            spyOn(questionModel, 'isValid').and.returnValue(true);
```

Comment:
It seems odd to me that the test "should state the question is invalid" to pass with this line added... specifically, see line 392.  Maybe I'm missing something>

Reply:
sorry, line 392 in the same file, I was referring to this test:

https://github.com/openmrs/openmrs-module-uicommons/blob/master/omod/src/test/webapp/resources/scripts/navigatorModels.js#L383

Specifically, on line 229, if I understand it correctly, we are mocking the questionModel 'isValid' method and telling it to return true, so I would assume line 392 would set isValid=true, yet 396 expects it to be false... so I would have expected the test starting on line 383 to be failing, but if it isn't I'm likely missing something.